airflow/providers/amazon/aws/hooks/rds.py

```diff
@@ -240,7 +240,7 @@ def get_db_instance_state(self, db_instance_id: str) -> str:
         try:
             response = self.conn.describe_db_instances(DBInstanceIdentifier=db_instance_id)
         except self.conn.exceptions.ClientError as e:
-            if e.response["Error"]["Code"] == "DBInstanceNotFoundFault":
+            if e.response["Error"]["Code"] == "DBInstanceNotFound":
```

Comment:
@AetherUnbound Thanks for creating the PR :)

Are there any version changes for SDK/Client/API used now vs earlier?

Reply:
Better to catch RDS.Client.exceptions.DBInstanceNotFoundFault (as `self.conn.exceptions.DBInstanceNotFoundFault) rather than generic ClientError and parse it.

Reply:
But this is weird because from [documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds/client/describe_db_instances.html) it is `DBInstanceNotFoundFault `

Reply:
@vincbeck Right, I mention this in the PR description - the error code itself within the `ClientError` is `DBInstanceNotFound`, but the exception is `DBInstanceNotFoundFault`. I'm all for catching a more specific exception, but the other functions in this file will probably need to be updated as well and I wanted to confirm that was the right way forward before making that change 🙂 

> Are there any version changes for SDK/Client/API used now vs earlier?

Not that I'm aware of, I think this has been an issue since we started using this DAG! Let me take a look at our logs and get back to you though.

Reply:
But [boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds/client/describe_db_instances.html) is different from [AWS API reference documentation](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBInstances.html#API_DescribeDBInstances_Errors):
- Boto3 documentation: `DBInstanceNotFoundFault`
- API reference documentation: `DBInstanceNotFound`

I trust your testing then and `DBInstanceNotFound` should be the good one 👍

Reply:
Response Error Code != Exception name

Some simple snippet for play with debugger

```python
import boto3
from botocore.exceptions import ClientError

session = boto3.session.Session(...) # do not forget add creds/profile or set appropriate ENV Vars
client = session.client("rds")


try:
    client.describe_db_instances(DBInstanceIdentifier="foo-bar-spam-egg")
except client.exceptions.DBInstanceNotFoundFault as ex:
    assert isinstance(ex, ClientError)
    assert isinstance(ex, client.exceptions.ClientError)
    raise
```

![image](https://github.com/apache/airflow/assets/3998685/6391325b-e045-45cf-b100-ac2ffaea535f)